src/plugins/GUITestBase/src/GTUtilsDashboard.cpp

```diff
@@ -146,6 +207,45 @@ bool GTUtilsDashboard::hasNotifications(HI::GUITestOpStatus &os) {
     return notificationsWidget->isVisible();
 }
 
+#define GT_METHOD_NAME "getNotifications"
+QList<GTUtilsDashboard::Notification> GTUtilsDashboard::getNotifications(GUITestOpStatus &os) {
+    using Type = Notification::Type;
+
+    const QString notificationsWidgetName = QStringLiteral("NotificationsDashboardWidget");
+    const auto tableLayout = qobject_cast<QGridLayout *>(
```

Comment:
Use findExactWidget so you do not need to check for nullptr.

Reply:
It won't work here

Reply:
It won't work here

Reply:
Ok, could you provide more details please why findExact won't work here?

Reply:
QLayout is not an inheritor of the QWidget class

I realized my mistake. I will make the code more readable